wgpu/src/lib.rs

```diff
@@ -2078,6 +2086,40 @@ impl Adapter {
     pub fn get_texture_format_features(&self, format: TextureFormat) -> TextureFormatFeatures {
         Context::adapter_get_texture_format_features(&*self.context, &self.id, format)
     }
+
+    /// Correleates presentation timestamps with a given user timestamp by generating a presentation timestamp
+    /// immediately after calling the user's timestamp function.
+    ///
+    /// When comparing completely opaque timestamp systems, we need a way of generating timestamps that signal
+    /// the exact same time. This function allows you to generate a timestamp (such as an [Instant]) at the
+    /// exact same time as a `PresentationTimestamp` is generated, allowing you to translate from one to another.
```

Comment:
This seems pretty complicated. I understand that we're trying to avoid blocking on acquiring wgpu's read locks for the adapter, but wouldn't it be simpler to just return a `PresentationTimestamp` and tell the user to call their own timestamp function immediately *after* this function returns?

> Note that this function may block waiting to acquire locks internally. To prevent this from introducing a delay between your timestamp and the WSI's timestamp, generate your timestamp immediately *after* this function returns. (Releasing locks never blocks.)

Reply:
Did some testing and this seems viable.

The difference between the current scheme and doing it after the timestamp is 0 to 300ns in release and 300 to 1500ns in debug. This is perfectly acceptable, especially considering the precision of timestamps may be way less than that anyway